linfa-clustering/src/gaussian_mixture/algorithm.rs

```diff
@@ -548,6 +550,48 @@ mod tests {
         assert_abs_diff_eq!(gmm.covariances(), &covars, epsilon = 1e-1);
     }
 
+    fn function_test_1d(x: &Array2<f64>) -> Array2<f64> {
+        let mut y = Array2::zeros(x.dim());
+        Zip::from(&mut y).and(x).apply(|yi, &xi| {
+            if xi < 0.4 {
+                *yi = xi * xi;
+            } else if xi >= 0.4 && xi < 0.8 {
+                *yi = 3. * xi + 1.;
+            } else {
+                *yi = f64::sin(10. * xi);
+            }
+        });
+        y
+    }
+
+    #[test]
+    fn test_zeroed_reg_covar_failure() {
+        let mut rng = Isaac64Rng::seed_from_u64(42);
+        let xt = Array2::random_using((50, 1), Uniform::new(0., 1.), &mut rng);
+        let yt = function_test_1d(&xt);
+        let data = stack(Axis(1), &[xt.view(), yt.view()]).unwrap();
```

Comment:
out of curiosity: that Cov(X, X^2) = 0 is obvious here for X ~ U(1, 0), but what's for the other two cases?